### ISSUE TITLE
[20220626] Weekly AI ArXiv 만담 - 56회차

### ISSUE THREAD

**Issue by jungwoo-ha** [jungwoo-ha/WeeklyArxivTalk#56]
### News
- Conferences
  - CVPR 2022: 6.19 ~ 24
  - EMNLP 2022: 모두들 수고 많으셨습니다.
- [E2V-SDE 논문 표절 이슈](https://twitter.com/e2v_sde_parody/status/1540087877308239874)

### ArXiv
- [Scaling Autoregressive Models for Content-Rich Text-to-Image Generation](https://arxiv.org/abs/2206.10789)
  - Google research에서 나온 또다른 Text-to-image generation 이름하여 Parti
  - 이미지 생성에 Diffusion model을 쓰지않고 Transformer S2S + ViT-VQGAN Tokenizer 활용 --> Autoregressive !
  - text encoder 학습에 BERT MLM loss로 C4 학습, LAION-400M, ALIGN일부, JFT-4B일부(SimVLM 통한 캡션) 등을 contrastive learning loss로 학습
  - 1024 해상도를 위해서 학습 파라미터 기반 Super-resolution upsampler 활용
  - Parti Prompt라는 벤치마크도 함께 공개
  - 모델의 크기(자본)가 이미지 생성에 어느정도 영향을 주는 지 적나라하게 확인 가능
![image](https://user-images.githubusercontent.com/11782739/175798973-5228b244-2543-40ba-8281-1bb94f4ab304.png)
![image](https://user-images.githubusercontent.com/11782739/175799015-ca23e12d-33f7-4389-ad99-33e2b4faa1d6.png)
![image](https://user-images.githubusercontent.com/11782739/175799172-af5c2091-2538-4865-9c96-d9ed6b904f66.png)
![image](https://user-images.githubusercontent.com/11782739/175799181-a05743b7-6241-4c7e-8e51-1e8c574d0d4e.png)

- [Rarity Score : A New Metric to Evaluate the Uncommonness of Synthesized Images](https://arxiv.org/abs/2206.08549)
  - 생성된 이미지가 얼마만큼 전형적이 않고 독창성이 있는지 평가하는 metric (from KAIST-NAVER Hypercreative AI Center)
  - 모델의 창의성을 평가하기 위해 출발한 연구
  - 기존 FID는 fidelity만 평가, LPIPS는 전반적인 분포의 다양성만 측정 가능
  - 이미지당 특이함을 나타내는 score이므로 개별 이미지가 얼마나 특이한지도 되고 생성이미지의 score분포를 통해 모델의 성능평가도 가능
  - 널리 쓰이는 truncation trick이 실제 rarity score에 영향을 주는 것을 확인
  - https://parti.research.google/
![image](https://user-images.githubusercontent.com/11782739/175794842-2c0963be-5bec-488f-8202-9ae8f2f2e705.png)
![image](https://user-images.githubusercontent.com/11782739/175794849-ad1c9419-72e4-45c7-816f-0fb146ff21d4.png)
![image](https://user-images.githubusercontent.com/11782739/175794857-9e47a759-fc25-4ab0-b80a-42ac850b2c3e.png)

- 흥미있는 연구
  - [OpenXAI: Towards a Transparent Evaluation of Model Explanations](https://arxiv.org/abs/2206.11104)
    - https://open-xai.github.io/
    - XAI 연구하시는 분들을 위한 성능 평가 벤치마크와 평가지표 프로토콜 정리해둠. 
  - [Few-Max: Few-Shot Domain Adaptation for Unsupervised Contrastive Representation Learning](https://arxiv.org/abs/2206.10137)
    - Contrastive self-supervised로 pretrain한 모델을 few-shot domain adaption 써먹을 때 (from UT Austin)
  - [Open-source language AI challenges big tech's models](https://www.nature.com/articles/d41586-022-01705-z)
    - https://bigscience.notion.site/BLOOM-BigScience-176B-Model-ad073ca07cdf479398d5f95d88e218c4
  - [Video PreTraining (VPT): Learning to Act by Watching Unlabeled Online Videos](https://arxiv.org/abs/2206.11795)
    - Minecraft playing video를 보고 학습하는 모델! (from OpenAI)
    - https://openai.com/blog/vpt/

**Comment by hollobit** [comment]
### Github Copilot 유료화 포함한 공식 서비스 시작  
https://github.com/github-copilot/signup
https://techcrunch.com/2022/06/21/copilot-githubs-ai-powered-programming-assistant-is-now-generally-available/
https://analyticsindiamag.com/6-month-review-github-copilot-elicits-mixed-reactions-from-developers/
>- 학생들과 검증된 오픈소스 컨트리뷰터들에게는 무료로 제공되지만 그외 개발자들에게는 월 10달러, 연간 기준으로는 100달러 비용이 부과
>- 공개된 소스코드 수십억 라인에서 훈련된 AI 모델 '코덱스'를 기반으로 기존 코드 맥락을 고려해 코드 라인들을 제시 
>- 프리뷰 기간 120만명 규모 개발자가 코파이럿에 가입. 현재 새로 쓰여지는 코드의 40%가 코파일럿이 제안했다고 
>- CoPilot 사용 5개월간의 경험담 기록 - https://medium.com/@buddi/github-copilot-after-using-5-months-2b06b79bb1f1

### Amazon launches CodeWhisperer, a GitHub Copilot-like AI pair programming tool
https://techcrunch.com/2022/06/23/amazon-launches-codewhisperer-its-ai-pair-programming-tool/
>- re:Mars conference에서 공개 
>- Java, JavaScript 및 Python 등 수십억 라인의 오픈 소스 코드와 자체 코드베이스, 공개 포럼에서 공개적으로 사용 가능한 문서 및 코드를 이용해 학습
>- 개발자용 IDE 지원 : Visual Studio Code, IntelliJ IDEA, PyCharm, WebStorm and Amazon’s own AWS Cloud 9. AWS Lambda Console도  곧 지원 예정

**Comment by ghlee0304** [comment]
### Arxiv (Audio and Speech Processing)

- [Nonwords Pronunciation Classification in Language Development Tests for Preschool Children](https://arxiv.org/pdf/2206.08058.pdf)
    - INTERSPEECH2022 /  Technische Hochschule Nurnberg Georg Simon Ohm(Germany), Intel Labs / 
    - Summary  
         - 어린이의 언어 발달이 연령에 적합한지 여부를 자동으로 평가하는 것을 목표로 하고 있음 
    - Key methods
    - Contributions
- [DRAFT: A Novel Framework to Reduce Domain Shifting in Self-supervised Learning and Its Application to Children’s ASR](https://arxiv.org/pdf/2206.07931.pdf)
- [NU-Wave 2: A General Neural Audio Upsampling Model for Various Sampling Rates](https://arxiv.org/pdf/2206.08545.pdf)
- [Avoid Overfitting User Specific Information in Federated Keyword Spotting](https://arxiv.org/pdf/2206.08864.pdf)
- [What can Speech and Language Tell us About the Working Alliance in Psychotherapy](https://arxiv.org/pdf/2206.08835.pdf)
- [On the Role of Spatial, Spectral, and Temporal Processing for DNN-based Non-linear Multi-channel Speech Enhancement](https://arxiv.org/pdf/2206.11181.pdf)
- [Conformer with dual-mode chunked attention for joint online and offline ASR](https://arxiv.org/pdf/2206.11157.pdf)
- [A Systematic Comparison of Phonetic Aware Techniques for Speech Enhancement](https://arxiv.org/pdf/2206.11000.pdf)
- [Radio2Speech: High Quality Speech Recovery from Radio Frequency Signals](https://arxiv.org/pdf/2206.11066.pdf)
- [Jointist: Joint Learning for Multi-instrument Transcription and Its Applications](https://arxiv.org/pdf/2206.10805.pdf)
- [Towards End-to-End Private Automatic Speaker Recognition](https://arxiv.org/pdf/2206.11750.pdf)
- [Efficient Transformer-based Speech Enhancement Using Long Frames and STFT Magnitudes](https://arxiv.org/pdf/2206.11703)
- [Adversarial Multi-Task Learning for Disentangling Timbre and Pitch in Singing Voice Synthesis](https://arxiv.org/pdf/2206.11558.pdf)
- [Towards Green ASR: Lossless 4-bit Quantization of a Hybrid TDNN System on the 300-hr Switchboard Corpus](https://arxiv.org/pdf/2206.11643.pdf)
- [Formant Estimation and Tracking using Probabilistic Heat-Maps](https://arxiv.org/pdf/2206.11632.pdf)

**Comment by kimyoungdo0122** [comment]
- News
    - Hugging face에서 [Dall-E mini 프롬프트](https://huggingface.co/spaces/dalle-mini/dalle-mini)를 써볼 수 있어요!
    - [craiyon](https://www.craiyon.com/) 사이트에서도 쓸 수 있습니다
    - Dall-e mini다보니 퀄리티가 아쉽고, 생성된 사람 얼굴은 많이 불쾌하실 수 있어요 ㅠ
    - 사전적으로 드물게 정의된 ['crungus'라는 단어에서 생성된 이미지](https://futurism.com/ai-nightmare-crungus)가 트위터를 떠들썩하게 했습니다
    - 손톱 밑의 때를 의미하지만 거의 사용되지 않는 단어에서 악마같은 형상의 이미지가 생성되었고 널리 퍼졌습니다
    - 보통의 어휘와 이미지로 학습되었다고 생각하기 어려운 경우인지라 더욱 이슈가 되었네요
    - 모 트위터유저가 찾아본 바로는 오래된 공포 게임에서 나온 몬스터라고 하는데 이것만으로 학습되었을까요?
    
![image](https://user-images.githubusercontent.com/42539153/175814816-3043052e-9648-47fb-ad84-80b53f5cc0fa.png)
![image](https://user-images.githubusercontent.com/42539153/175814831-73758337-5546-4e28-8448-42cf5a4f7b64.png)
![image](https://user-images.githubusercontent.com/42539153/175814842-79406a43-7064-4edd-8abc-383aa1ddeb9e.png)
![image](https://user-images.githubusercontent.com/42539153/175814944-a677e513-762f-4a7e-9585-c682c68a63da.png)
![image](https://user-images.githubusercontent.com/42539153/175815038-158f679b-3f24-4fe0-99b3-aae1359ce325.png)

**Comment by veritas9872** [comment]
**Tracing with Primitives: TorchDynamo & nvFuser**

PyTorch 라이브러리의 미래 진행 방향이 Julia와 매우 유사한 방향으로 진행되는 것으로 PyTorch Dev Blog에 공지되었습니다.

지금까지 딥러닝 라이브러리의 주요 패러다임은 low-level CUDA C/C++로 핵심 코드를 작성하고 Python으로 high-level wrapper를 작성하는 방식으로만 진행이 되어오되 eager execution을 통해 사용자 편의성을 중시해오고 최적화를 위해서는 flexibility를 희생시킨 graph mode를 사용해서 JIT Compile을 진행해야만 했습니다.

하지만 현재 방식은 새로운 operation을 정의할 때마다 low-level kernel을 작성해야해서 현재 PyTorch에는 2,000개가 넘는 operator가 있다고 합니다.

이런 문제점을 해결하기 위해 primitive operation을 정의하고 Python에서 interpreter가 bytecode를 받는 단계에서 bytecode를 수정함으로써 symbolic kernel fusion을 실현하는 방향으로 나아갈 것이라고 합니다. 이런 방법의 장점은 추후 high-level에서 Python으로 새로운 프로그램을 작성하더라도 여러 hardware accelerator에서 최적화된 kernel이 실행된다는 점입니다.

이미 90% 이상의 repository에서 accuracy가 100%인 것을 발표했는데 추후 연구개발을 진행하는데 많은 도움이 될 것 같습니다.

PyTorch Dev Blog에 관련 여러 포스트가 있는데 아래 정리해드립니다.

TorchDynamo GitHub: https://github.com/pytorch/torchdynamo

Where PyTorch is Headed: https://dev-discuss.pytorch.org/t/where-we-are-headed-and-why-it-looks-a-lot-like-julia-but-not-exactly-like-julia/276
[Tracing 0](https://dev-discuss.pytorch.org/t/tracing-with-primitives-update-0/577)
[Tracing 1](https://dev-discuss.pytorch.org/t/tracing-with-primitives-update-1-nvfuser-and-its-primitives/593)
[Tracing 2](https://dev-discuss.pytorch.org/t/tracing-with-primitives-update-2/645)
[NVFuser 1](https://www.nvidia.com/en-us/on-demand/session/gtcspring21-s31952/)
[NVFuser 2](https://events.rainfocus.com/widget/nvidia/gtcspring2022/sessioncatalog/session/1638827482276001DLEK)

**DeepMind Educational Resources**

GitHub: https://github.com/deepmind/educational

딥마인드에서 머신러닝 및 딥러닝을 설명하기 위한 Colab이 오픈소스되었습니다. 비전문가에게 딥러닝을 설명하실 때 많은 도움이 될 것 같습니다.

**OmniXAI**

GitHub: https://github.com/salesforce/OmniXAI

Salesforce에서 설명가능한 인공지능을 위한 OmniXAI 라이브러리를 공개했습니다. PyTorch의 Captum과 함께 연구자들에게 많은 도움이 될 것 같습니다.

**Unified-IO: A Unified Model for Vision, Language, and Multi-Modal Tasks**

Allen Institute for AI 에서 Unified IO라는 vision/language general-purpose model을 공개했습니다.

Arxiv: https://arxiv.org/abs/2206.08916

Blog: https://unified-io.allenai.org/

Medium: https://ai2.medium.com/9c0ec7fe1e43

**Comment by jwlee-ml** [comment]
<홍보 한가지 하겠습니다 :) >
[PR12 5기 멤버 모집: 6명] #PR12 - 7/10(일) 23시59분59초까지!!

안녕하세요. PR12에서 5기 멤버를 추가로 6명 모집합니다
PR12는 AI관련 논문을 리뷰하는 모임으로
2017년 4월부터 6년째 매주 일요일 밤 10시부터 Zoom을 이용하여 두 분씩 돌아가면서 온라인으로 발표를 하고
발표를 모두 녹화하여 Youtube에 공유하고 있습니다.

작년 3월 7일 첫 발표를 시작한 PR12 Season 4가 어느덧 100편의 논문 리뷰를 앞드고 있습니다(현재 392편 리뷰 완료)
그동안 고생해주신 4기 멤버들께 깊은 감사를 드립니다
이제 새롭게 Season 5를 함께해주실 5기 멤버를 모집합니다!!! 

PR12를 함께하시면 함께하시는 훌륭하신 리뷰어 분들과 네트워킹도 할 수 있고
무엇보다 1년 후에 많이 성장한 스스로를 발견하실 수 있습니다
다들 아시죠? 선지원 후고민(or 노고민)!

* 자격요건
- 나이, 성별, 국적, 전공, 직업 불문
- 영어 논문읽기에 큰 불편이 없으신분
- 딥러닝에 대한 이해가 있으신분
- 6개월 이상 매주일 한번씩 끈기 있게 참여하실수 있으신분 (일요일 밤에 시간이 가능하신 분)
- 본인이 딥러닝 관련학회 논문을 제출해 보셨거나 기존의 논문 알고리즘등을 구현해보신분 (Plus)
- 슬라이드 잘 만드시고 발표잘하시는 분 (Plus)
* 지원 방법 (7월 10일 23:59분59초 AOE 마감)
- 희망하시는 논문 하나를 읽고, 발표슬라이드를 만드신후, 5분정도 발표하는 모습을 (스크린 리코딩) youtube에 올리시고 간단한 자기 소개와 함께 jwlee.ml25@gmail.com 으로 link를 보내주시면 됩니다.
- 본인이 이미 구현한 딥러닝 모델들이 있으시면 github 주소등 같이 보내 주세요. (plus)
* 기타 문의: jwlee.ml25@gmail.com

정말 딥러닝 초 고수가 될수 있는 PR12에 많은 관심 부탁드립니다.
(발표된 논문슬라이드와 영상은 Youtube에모두 공개하고 있습니다!)

<PR12 다시보기>
Season 1 - https://www.youtube.com/watch?v=auKdde7Anr8&list=PLWKf9beHi3Tg50UoyTe6rIm20sVQOH1br
Season 2 - https://www.youtube.com/watch?v=FfBp6xJqZVA&list=PLWKf9beHi3TgstcIn8K6dI_85_ppAxzB8
Season 3 - https://www.youtube.com/watch?v=D-baIgejA4M&list=PL_skMddDjnzq1wDI3t2cH9hlK6wBBapeA
Season 4 - https://www.youtube.com/watch?v=az-OV47oKvA&list=PL0o99tZwBlrMV3QsZ4O79KjMHDhAJpAdW
